### ISSUE TITLE
Re-evaluate usage of Snorenotify

### ISSUE THREAD

**Issue by nurupo** [qTox/qTox#6238]
I was bumping Windows dependencies and noticed [Snorenotify being added](https://github.com/qTox/qTox/commit/7409e6b4cc584f1d1737e180b4066be371303801). 

I have looked into Snorenotify and found several issues with it:

1. Snorenotify is unmaintained.

    Its repo has gained the next text in August:

    >[Snorenotify is unmaintained, it is recommended to use KNotifications instead. If you only care for Windows notification use https://invent.kde.org/libraries/snoretoast](https://github.com/KDE/snorenotify/commit/8b64f8a1cdc8cb6abe3a59ffa52f108342433c63)

2. Snorenotify depends on [SnoreToast](https://github.com/KDE/snoretoast) on Windows, which can be built only with MSVC.

    a. According to SnoreToast's maintainer (TheOneRing aka the2ring on #kde-windows), SnoreToast can only be built using MSVC, it won't build with MinGW-W64. This means that we can't build it on Debian, thus we unable to use Snorenotify, at least not on Windows.

    b. You might have noticed that despite (a) we do somehow build and use it, you can see the build script running `cp` on SnoreToast.exe, i.e. the binary is being produced somehow, and our CI clearly doesn't fail even though it should be impossible to build with MinGW-W64. That's because the tarball includes a pre-built `./src/plugins/backends/snoretoast/SnoreToast.exe` (!), which is a big no-no as it has been our policy to not use any pre-built binaries and instead build them ourselves from the trusted version of the source code. (Well, we do violate this policy by downloading NSIS's ShellExecAsUser.dll plugin, which needs to be addressed, perhaps in a separate issue).

    So we need to see if SnoreToast source code and cmake can be patched so that it builds with MingW-W64, and if not -- drop Snorenotify/SnoreToast on Windows.

Do we want to revert https://github.com/qTox/qTox/commit/7409e6b4cc584f1d1737e180b4066be371303801 or keep using it knowing that it uses the pre-built `SnoreToast.exe`?

**Comment by sudden6** [comment]
I agree, snorenotify was not a good choice, it was already on the path of deprecation when I implemented it, but I had hope.

It seems Qt's mechanism for notifications is `QSystemTrayIcon` and we should give that a try.

**Comment by anthonybilinski** [comment]
Linking to https://github.com/qTox/qTox/pull/6296.

**Comment by nurupo** [comment]
Thanks for the update. Looks like KNotifications can use SnoreToast for Windows too, [however it seems to be optional](https://github.com/KDE/knotifications/blob/v5.78.0/CMakeLists.txt#L74-L82), so that solves the issue with SnoreToast, in a way.

Btw, I have asked what exactly makes SnoreToast not work with MingW-W64 on #kde-windows, since I wasn't given the actual reason the last time, and turns out it's because mingw is missing WinRT APIs/headers that SnoreToast needs.
